core/src/main/java/org/apache/cxf/helpers/HttpHeaderHelper.java

```diff
@@ -112,8 +111,8 @@ public static String mapCharset(String enc, String deflt) {
         }
         // Charsets can be quoted. But it's quite certain that they can't have escaped quoted or
         // anything like that.
-        enc = charsetPattern.matcher(enc).replaceAll("").trim();
-        if ("".equals(enc)) {
+        enc = enc.replace('"', ' ').replace('\'', ' ').trim();
```

Comment:
Is there a reason to replace the previous behaviour (empty String) with a whitespace? Also, it's only replacing the first instance of each? I'm just wondering if there might be an attack where an attacker could force using a charset by escaping twice in a row with the new code.

Reply:
char replacement performed without Pattern usage. its required only in case of quotes and removed by existing trim. result value used as argument for Character.forName which is safe and throws exception on any wrong input

Reply:
LGTM thanks